File: tests/cases/check/max-class-fanout/csharp/inputs/src/Orders/Pricing.cs
namespace Orders;

// Four distinct references, with `Cart` named twice - fanout counts DISTINCT
// targets, so this is 4, not 5. Under the limit.
public class PriceCalculator
{
    public Cart Cart { get; }
    public Cart Basket { get; }
    public PriceTable Pricing { get; }

    public PriceCalculator(Cart cart, Cart basket, PriceTable pricing)
    {
        Cart = cart;
        Basket = basket;
        Pricing = pricing;
    }

    public decimal Quote(Inventory inv, ShippingCalculator ship) { return 0m; }
}

File: tests/cases/check/max-class-fanout/python/inputs/src/orders/pricing.py
from orders.collaborators import Cart, Inventory, PriceTable, ShippingCalculator


class PriceCalculator:
    """Four distinct references, and `Cart` twice - fanout counts DISTINCT
    targets, so this is 4, not 5. Under the limit."""

    def __init__(self, cart: Cart, basket: Cart, pricing: PriceTable) -> None:
        self.cart: Cart = cart
        self.basket: Cart = basket
        self.pricing: PriceTable = pricing

    def quote(self, inv: Inventory, ship: ShippingCalculator) -> float:
        return 0.0
